File: VB/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System.Web.Mvc
Imports E4425.Models
Imports DevExpress.Web.Mvc
Imports System.Linq
Imports System

Namespace E4425.Controllers
    Public Class HomeController
        Inherits Controller
        Private entity As New WorldCitiesEntities()
        Public Function Index() As ActionResult
            Return View(entity.Customers.ToList())
        End Function
        Public Function GridViewPartial() As ActionResult
            Return PartialView(entity.Customers.ToList())
        End Function
        Public Function GridViewEditPartial(ByVal customerInfo As Customer) As ActionResult
            If ModelState.IsValid Then
                Try
                    entity.Customers.Attach(customerInfo)
                    Dim entry = entity.Entry(customerInfo)
                    entry.Property(Function(e) e.CityId).IsModified = True
                    entry.Property(Function(e) e.CountryId).IsModified = True
                    entry.Property(Function(e) e.CustomerName).IsModified = True
                    ' uncomment the next line to enable database updates
                    ' entity.SaveChanges();
                Catch e As Exception
                    ViewData("EditError") = e.Message
                End Try
            Else
                ViewData("EditError") = "Please, correct all errors."
            End If
            Return PartialView("GridViewPartial", entity.Customers.ToList())
        End Function
        Public Function GridViewInsertPartial(ByVal customerInfo As Customer) As ActionResult
            If ModelState.IsValid Then
                Try
                    entity.Customers.Add(customerInfo)
                    ' uncomment the next line to enable database updates
                    ' entity.SaveChanges();
                Catch e As Exception
                    ViewData("EditError") = e.Message
                End Try
            Else
                ViewData("EditError") = "Please, correct all errors."
            End If
            Return PartialView("GridViewPartial", entity.Customers.ToList())
        End Function
        Public Function GridViewDeletePartial(ByVal CustomerId As Integer) As ActionResult
            If ModelState.IsValid Then
                Try
                    entity.Customers.Remove(entity.Customers.Find(CustomerId))
                    'uncomment the next line to enable database updates
                    'entity.SaveChanges();
                Catch e As Exception
                    ViewData("EditError") = e.Message
                End Try
            Else
                ViewData("EditError") = "Please, correct all errors."
            End If
            Return PartialView("GridViewPartial", entity.Customers.ToList())
        End Function
        Public Function ComboBoxCountryPartial() As ActionResult
            Return GridViewExtension.GetComboBoxCallbackResult(ComboBoxPropertiesProvider.Current.CountryComboBoxProperties)
        End Function
        Public Function ComboBoxCityPartial() As ActionResult
            Return GridViewExtension.GetComboBoxCallbackResult(ComboBoxPropertiesProvider.Current.CityComboBoxProperties)
        End Function
    End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System;
using System.Web.Mvc;
using E4425.Models;
using DevExpress.Web.Mvc;
using System.Linq;
namespace E4425.Controllers
{
    public class HomeController : Controller
    {
        WorldCitiesEntities entity = new WorldCitiesEntities();
        public ActionResult Index()
        {
            return View(entity.Customers.ToList());
        }
        public ActionResult GridViewPartial()
        {
            return PartialView(entity.Customers.ToList());
        }
        public ActionResult GridViewEditPartial(Customer customerInfo)
        {
            if (ModelState.IsValid)
            {
                try
                {
                    entity.Customers.Attach(customerInfo);
                    var entry = entity.Entry(customerInfo);
                    entry.Property(e => e.CityId).IsModified = true;
                    entry.Property(e => e.CountryId).IsModified = true;
                    entry.Property(e => e.CustomerName).IsModified = true;
                    // uncomment the next line to enable database updates
                    //   entity.SaveChanges();
                }
                catch (Exception e)
                {
                    ViewData["EditError"] = e.Message;
                }
            }
            else
                ViewData["EditError"] = "Please, correct all errors.";
            return PartialView("GridViewPartial", entity.Customers.ToList());
        }
        public ActionResult GridViewInsertPartial(Customer customerInfo)
        {
            if (ModelState.IsValid)
            {
                try
                {
                    entity.Customers.Add(customerInfo);
                    // uncomment the next line to enable database updates
                    // entity.SaveChanges();
                }
                catch (Exception e)
                {
                    ViewData["EditError"] = e.Message;
                }
            }
            else
                ViewData["EditError"] = "Please, correct all errors.";
            return PartialView("GridViewPartial", entity.Customers.ToList());
        }
        public ActionResult GridViewDeletePartial(int CustomerId)
        {
            if (ModelState.IsValid)
            {
                try
                {
                    entity.Customers.Remove(entity.Customers.Find(CustomerId));
                    // uncomment the next line to enable database updates
                    //entity.SaveChanges();
                }
                catch (Exception e)
                {
                    ViewData["EditError"] = e.Message;
                }
            }
            else
                ViewData["EditError"] = "Please, correct all errors.";
            return PartialView("GridViewPartial", entity.Customers.ToList());
        }
        public ActionResult ComboBoxCountryPartial()
        {
            return GridViewExtension.GetComboBoxCallbackResult(ComboBoxPropertiesProvider.Current.CountryComboBoxProperties);
        }
        public ActionResult ComboBoxCityPartial()
        {
            return GridViewExtension.GetComboBoxCallbackResult(ComboBoxPropertiesProvider.Current.CityComboBoxProperties);
        }
    }

}